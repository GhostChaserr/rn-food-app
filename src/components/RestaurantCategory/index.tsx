import React, { FC } from 'react'
import { ImageSourcePropType } from 'react-native'
import {
  StyledCategoryButton,
  StyledCategoryWrapper,
  StyledCategoryIcon,
  StyledCategoryTextWrapper,
  StyledCategoryText
} from './styles'

export type RestaurantCategoryT = {
  id: number
  icon: ImageSourcePropType
  name: string
}

type RestaurantCategoryProps = {
  id: number
  selected: boolean
  name: string
  icon: ImageSourcePropType
  onCategorySelect: (category: RestaurantCategoryT) => void
}

const RestaurantCategory: FC<RestaurantCategoryProps> = (props) => {
  return (
    <StyledCategoryButton
      isSelected={props.selected}
      onPress={() =>
        props.onCategorySelect({
          id: props.id,
          icon: props.icon,
          name: props.name
        })
      }
    >
      <StyledCategoryWrapper>
        <StyledCategoryIcon source={props.icon} />
      </StyledCategoryWrapper>
      <StyledCategoryTextWrapper>
        <StyledCategoryText isSelected={props.selected}>
          {props.name}
        </StyledCategoryText>
      </StyledCategoryTextWrapper>
    </StyledCategoryButton>
  )
}

export default RestaurantCategory
