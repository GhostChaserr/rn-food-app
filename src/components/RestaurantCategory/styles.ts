import styled from 'styled-components/native'

export const StyledCategoryButton = styled.TouchableOpacity<{ isSelected: boolean }>`
  padding: 10px;
  padding-bottom: 20px;
  background-color: ${({ isSelected }) => isSelected ? '#FC6D3F;': 'white;' } 
  border-radius: 30px;
  align-items: center;
  justify-content: center;
  margin-right: 10px;
`

export const StyledCategoryWrapper = styled.View`
  width: 50px;
  height: 50px;
  align-items: center;
  justify-content: center;
  background-color: white;
  border-radius: 30px;
`


export const StyledCategoryIcon = styled.Image`
  resize-mode: contain;
  width: 35px;
  height: 35px;
`


export const StyledCategoryTextWrapper = styled.View`
  margin-top: 10px;
  align-items: center;
  justify-content: center;
`

export const StyledCategoryText = styled.Text<{ isSelected: boolean }>`
  font-size: 11px;
  color: ${({ isSelected }) => isSelected ? 'white;': 'black;' } 
  font-weight: bold;
`