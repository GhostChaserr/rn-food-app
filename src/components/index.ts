export { default as RestaurantPreview } from './RestaurantPreview'
export { default as RestaurantCategory } from './RestaurantCategory'