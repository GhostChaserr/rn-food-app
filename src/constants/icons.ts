export const back = require('src/assets/icons/back.png')
export const car = require('src/assets/icons/car.png')
export const cutlery = require('src/assets/icons/cutlery.png')
export const donut = require('src/assets/icons/donut.png')
export const drink = require('src/assets/icons/drink.png')
export const fire = require('src/assets/icons/fire.png')
export const fries = require('src/assets/icons/fries.png')
export const hamburger = require('src/assets/icons/hamburger.png')
export const hotdog = require('src/assets/icons/hotdog.png')
export const like = require('src/assets/icons/like.png')
export const list = require('src/assets/icons/list.png')
export const location = require('src/assets/icons/location.png')
export const master_card = require('src/assets/icons/mastercard.png')
export const nearby = require('src/assets/icons/nearby.png')
export const noodle = require('src/assets/icons/noodle.png')
export const pin = require('src/assets/icons/pin.png')
export const pizza = require('src/assets/icons/pizza.png')
export const red_pin = require('src/assets/icons/red-pin.png')
export const rice_bowl = require('src/assets/icons/rice-bowl.png')
export const salad = require('src/assets/icons/salad.png')
export const search = require('src/assets/icons/search.png')
export const basket = require('src/assets/icons/shopping-basket.png')
export const star = require('src/assets/icons/star.png')
export const sushi = require('src/assets/icons/sushi.png')
export const user = require('src/assets/icons/user.png')

export default {
  back,
  car,
  cutlery,
  donut,
  drink,
  fire,
  fries,
  hamburger,
  hotdog,
  like,
  list,
  location,
  master_card,
  nearby,
  noodle,
  pin,
  pizza,
  red_pin,
  rice_bowl,
  salad,
  search,
  basket,
  star,
  sushi,
  user
}
