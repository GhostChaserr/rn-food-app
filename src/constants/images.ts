const avatar_1 = require('src/assets/images/avatar-1.jpg')
const avatar_2 = require('src/assets/images/avatar-2.jpg')
const avatar_3 = require('src/assets/images/avatar-3.jpg')
const avatar_4 = require('src/assets/images/avatar-4.jpg')
const avatar_5 = require('src/assets/images/avatar-5.jpg')
const baked_fries = require('src/assets/images/baked-fries.jpg')
const burger_restaurant_1 = require('src/assets/images/burger-restaurant.jpg')
const burger_restaurant_2 = require('src/assets/images/burger-restaurant-2.jpg')
const chicago_hot_dog = require('src/assets/images/chicago-hot-dog.jpg')
const crispy_chicken_burger = require('src/assets/images/crispy-chicken-burger.jpg')
const fries_restaurant = require('src/assets/images/fries-restaurant.jpg')
const hawaiian_pizza = require('src/assets/images/hawaiian-pizza.jpg')
const honey_mustard_chicken_burger = require('src/assets/images/honey-mustard-chicken-burger.jpg')
const hot_dog_restaurant = require('src/assets/images/hot-dog-restaurant.jpg')
const ice_kacang = require('src/assets/images/ice-kacang.jpg')
const japanese_restaurant = require('src/assets/images/japanese-restaurant.jpg')
const kek_lapis_shop = require('src/assets/images/kek-lapis-shop.jpg')
const kek_lapis = require('src/assets/images/kek-lapis.jpg')
const kolo_mee = require('src/assets/images/kolo-mee.jpg')
const nasi_briyani_mutton = require('src/assets/images/nasi-briyani-mutton.jpg')
const nasi_lemak = require('src/assets/images/nasi-lemak.jpg')
const noodle_shop = require('src/assets/images/noodle-shop.jpg')
const pizza_restaurant = require('src/assets/images/pizza-restaurant.jpg')
const pizza = require('src/assets/images/pizza.jpg')
const salad = require('src/assets/images/salad.jpg')
const sarawak_laksa = require('src/assets/images/sarawak-laksa.jpg')
const sushi = require('src/assets/images/sushi.jpg')
const teh_c_peng = require('src/assets/images/teh-c-peng.jpg')
const tomato_pasta = require('src/assets/images/tomato-pasta.jpg')

export default {
  avatar_1,
  avatar_2,
  avatar_3,
  avatar_4,
  avatar_5,
  baked_fries,
  burger_restaurant_1,
  burger_restaurant_2,
  chicago_hot_dog,
  crispy_chicken_burger,
  fries_restaurant,
  hawaiian_pizza,
  honey_mustard_chicken_burger,
  hot_dog_restaurant,
  ice_kacang,
  japanese_restaurant,
  kek_lapis_shop,
  kek_lapis,
  kolo_mee,
  nasi_briyani_mutton,
  nasi_lemak,
  noodle_shop,
  pizza_restaurant,
  pizza,
  sarawak_laksa,
  sushi,
  teh_c_peng,
  tomato_pasta
}
