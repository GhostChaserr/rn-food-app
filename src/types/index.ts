import { ImageSourcePropType } from "react-native"

export type RestaurantMenuT = {
  menuId: number,
  name: string
  photo: ImageSourcePropType
  description: string
  calories: number,
  price: number
}

export type RestaurantPreviewT = {
  id: number
  name: string
  rating: number
  categories: string[]
  photo: ImageSourcePropType
  duration: string
  menu: RestaurantMenuT[]
}
