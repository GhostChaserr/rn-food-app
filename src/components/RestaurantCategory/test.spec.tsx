import React from 'react'
import TestRenderer from 'react-test-renderer';
import { icons, images } from 'src/constants'

import RestaurantCategory, { RestaurantCategoryT } from './index'

describe('Testing restaurant category component', () => {
  it('Should render as selected', () => {
    const testData: RestaurantCategoryT = {
      id: 1,
      name: 'Modern Fries',
      icon: icons.fries
    }
    const tree = TestRenderer
      .create(
        <RestaurantCategory
          onCategorySelect={(ct) => console.log(ct)}
          selected={true}
          id={testData.id}
          icon={testData.icon}
          name={testData.name}
        />
      )
    expect(tree).toMatchSnapshot();
  })

  it('Should render as not selected', () => {
    const testData: RestaurantCategoryT = {
      id: 1,
      name: 'Modern Fries',
      icon: icons.fries
    }
    const tree = TestRenderer
      .create(
        <RestaurantCategory
          onCategorySelect={(ct) => console.log(ct)}
          selected={false}
          id={testData.id}
          icon={testData.icon}
          name={testData.name}
        />
      )
      expect(tree).toMatchSnapshot();
  })
})
