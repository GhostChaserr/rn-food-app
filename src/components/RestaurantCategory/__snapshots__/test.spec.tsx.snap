// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing restaurant category component Should render as not selected 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 30,
      "justifyContent": "center",
      "marginRight": 10,
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 30,
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 35,
            "resizeMode": "contain",
            "width": 35,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 10,
        },
      ]
    }
  >
    <Text
      isSelected={false}
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 11,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Modern Fries
    </Text>
  </View>
</View>
`;

exports[`Testing restaurant category component Should render as selected 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FC6D3F",
      "borderRadius": 30,
      "justifyContent": "center",
      "marginRight": 10,
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 30,
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 35,
            "resizeMode": "contain",
            "width": 35,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 10,
        },
      ]
    }
  >
    <Text
      isSelected={true}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 11,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Modern Fries
    </Text>
  </View>
</View>
`;
